pkg/resmap/resmap.go

```diff
@@ -194,8 +194,14 @@ func (m *resWrangler) Resources() []*resource.Resource {
 func (m *resWrangler) Append(res *resource.Resource) error {
 	id := res.CurId()
 	if r := m.GetMatchingResourcesByCurrentId(id.Equals); len(r) > 0 {
-		return fmt.Errorf(
-			"may not add resource with an already registered id: %s", id)
+		// This allows diamond import of resources.
+		// Only errors if the resources are different because they have
+		// been loaded from a different source.
+		if !res.Equals(r[0]) {
```

Comment:
Why is `r[0]` special?  `GetMatchingResourcesByCurrentId` may return multiple resources.

Reply:
The structure of the list and the current Append function is ensuring that there is only one in the list.
For piece of mind, will add a test returning an error if the list is more than 1.